storage/client_test.go

```diff
@@ -217,66 +218,32 @@ func (m *multiTestContext) Stop() {
 
 // rpcSend implements the client.rpcSender interface. This implementation of "rpcSend" is
 // used to multiplex calls between many local senders in a simple way; It sends
-// the request to each localSender of a multiTestContext in order, stopping if
-// the request succeeds or any error other than RangeKeyMismatch is returned.
-//
-// TODO(mrtracy): remove once #2141 is merged and multiTestContext begins using
-// DistSender. This simple implementation will likely be incorrect in some
-// untested cases and is a temporary measure until DistSender is hooked up.
-func (m *multiTestContext) rpcSend(_ rpc.Options, _ string, _ []net.Addr,
+// the request to multiTestContext's localSenders specified in addrs. The request is
+// sent in order until no error is returned.
+func (m *multiTestContext) rpcSend(_ rpc.Options, _ string, addrs []net.Addr,
 	getArgs func(addr net.Addr) gogoproto.Message,
 	getReply func() gogoproto.Message, _ *rpc.Context) ([]gogoproto.Message, error) {
-	maxRetries := len(m.senders) * 2
-	// Iterate over all stores in the test context, looking for a suitable
-	// replica. In the case of a RangeKeyMismatchError, retry the request on the
-	// next store. In the case of a NotLeaderError, attempt the new leader next.
-	// Because NotLeaderErrors will adjust the index of our loop, a second
-	// "maxRetries" count is used to bound the number of retries.
-	nextIdx := 0
 	var call proto.Call
-	for total := 0; total < maxRetries; total++ {
+	for _, addr := range addrs {
 		call = proto.Call{
 			Args:  getArgs(nil /* net.Addr */).(proto.Request),
 			Reply: getReply().(proto.Response),
 		}
-		if err := client.SendCall(m.senders[nextIdx], call); err != nil {
-			switch err := err.(type) {
-			case *proto.RangeKeyMismatchError:
-				// Try the next localSender if this localSender did not have the
-				// requested range.
-				nextIdx++
-				nextIdx %= len(m.senders)
-				continue
-			case *proto.NotLeaderError:
+		// Node ID is encoded in the port number.
+		resolvedAddr, err := addr.(util.UnresolvedAddr).Resolve()
```

Comment:
I don't think there's any need to go through `Resolve()`. You can get the original string with `addr.String()`, so it can just be the port number with no `127.0.0.1:` (and put "local" instead of "tcp" in the `network` field).